### ISSUE TITLE
Timer on the invoice slows down when switching to other browser tabs (Safari)

### ISSUE THREAD

**Issue by raindogdance** [btcpayserver/btcpayserver#130]
I noticed that when I create a new invoice and then switch to another tab in my browser, the timer on the invoice slows down considerably, barely counting down at all.  

It seems like a Safari specific issue, because I couldn't recreate it with Brave and Chrome browsers (Safari 11.1 here). Also, the progress bar is showing up correctly, only the timer countdown is affected (as visible on the screenshot: Safari on the left, Brave on the right).

![2018-04-18 16 49 04](https://user-images.githubusercontent.com/35657440/38943094-73ea5a6a-4328-11e8-89d7-ee29779e188e.png)

**Comment by Eskyee** [comment]
I just tested this on my Safari 11.1  on my Apple Macbook
two times plus tested with chrome at the same time, 
macOS High Sierra 10.13.4
can't see any problem with the timers on my system in both browsers  ... 
?? don't know if thats any help..
also I using a 15 min timer..

**Comment by lepipele** [comment]
@raindogdance can you run that pull request locally and test if it works: https://github.com/btcpayserver/btcpayserver/pull/139 ?

@NicolasDorier if we can push this version to some staging environment for test, that could also be the way for test before pushing to production.

**Comment by raindogdance** [comment]
@lepipele I'm afraid this is a bit above my current capabilities, I'm not really a code tinkerer, sorry! :) Happy to help any way I can though, so if I can test just the invoice itself (I don't have btcpay installed, just was trying to pay an invoice when the issue occured) or something else, I'd do gladly :)

**Comment by lepipele** [comment]
OK, I'll see with @NicolasDorier if he can give me access to push staging changes like this to test environment like http://btcpay-server-testnet.azurewebsites.net/ so we can have external people like you test new releases before hitting production.

If not, he'll test locally and then we go: https://i.imgur.com/y7Hm9.jpg

**Comment by raindogdance** [comment]
I just registered/set up for testing and tried with two new invoices, and the timer is synchronous in both Safari and Brave! :) 

Although strangely enough, I cannot recreate the issue in the OP anymore (tried on @NicolasDorier tipping page as well as on bitcoinshirt.co).. So the code change has already been implemented for all BTCPay users, I assume? Otherwise, Safari had a bigger update yesterday, so perhaps that had smth to do with it, too.
